application/plugin_base/item_list.py

```diff
@@ -1647,6 +1665,19 @@ def destroy_providers(self):
 		for path, provider in self._providers.items():
 			provider.release_archive_handle()
 
+	def create_intermediate_archive_provider(self, path=None):
```

Comment:
Hi Mladen

Two scenarios:

1) left panel has archive open at /home/foo/bar.zip/subdir1, this contains
subdir2, CTRL-RIGHT is pressed and right panel doesn't have a ZipProvider
for archive yet. This method is called with
/home/foo/bar.zip/subdir1/subdir2 and creates a ZipProvider for
/home/foo/bar.zip such that /subdir1/subdir2 can be displayed in the right
panel

2) an archive was open in one panel before closing Sunflower at
/home/foo/bar.zip/subdir1 - again a ZipProvider for /home/foo/bar.zip is
needed in the panel after restart  before /home/foo/bar.zip/subdir1 can be
displayed.

I hope this makes it a bit clearer - one related change is in
zip_provider.set_archive_handle() to populate zip cache immediatley

Maybe this is not the most elegant way to handle the requirement - let me
know!

Cheers
Christian


On Sat, Feb 7, 2015 at 6:58 PM, Mladen Mijatov <notifications@github.com>
wrote:

> In application/plugin_base/item_list.py
> <https://github.com/MeanEYE/Sunflower/pull/65#discussion_r24291768>:
>
> > @@ -1647,6 +1665,19 @@ def destroy_providers(self):
> >  		for path, provider in self._providers.items():
> >  			provider.release_archive_handle()
> >
> > +	def create_intermediate_archive_provider(self, path=None):
>
> Can you explain purpose of this method?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/MeanEYE/Sunflower/pull/65/files#r24291768>.
>